### PR TITLE
Implement more selective remote position caching for audiobooks

### PR DESCRIPTION
**Pull request by mauricecarrier7**
**What's this do?**
- Throttles listening position save calls to server, saves once every 15 seconds while playing
- saves location on play, pause, seek, skip and chapter selection
- resolves TOC audiobook location server syncing issue

**Why are we doing this? (w/ Notion link if applicable)**
https://www.notion.so/lyrasis/iOS-Save-listening-position-less-frequently-6445ef4deb424103b2ec43e77541c418

**How should this be tested? / Do these changes have associated tests?**
- Play an audiobook, monitor console to evaluate time between location saves
- press play, notice save operation
- press pause, notice save operation
- seek, notice save operation
- scrub, notice save operation
- select chapter from spine, notice save operation

**Dependencies for merging? Releasing to production?**
https://github.com/ThePalaceProject/ios-audiobooktoolkit/pull/34/files

**Does this include changes that require a new Palace build for QA?**
Yes

**Has the application documentation been updated for these changes?**
N/A

**Did someone actually run this code to verify it works?**
@mauricecarrier7

### CODE REVIEW

File: Palace/Book/UI/TPPBookCellDelegate.m [hunk 1]
```diff
@@ -40,6 +39,8 @@ @interface TPPBookCellDelegate () <RefreshDelegate>
 
 @implementation TPPBookCellDelegate
 
+static int ServerUpdateDelay = 15;
```
Comment (vladimirfedorov) [inline Palace/Book/UI/TPPBookCellDelegate.m:42]: It's a variable, would be better to start with a lower case s

File: Palace/Book/UI/TPPBookCellDelegate.m [hunk 2]
```diff
@@ -500,8 +503,11 @@ - (void)pollAudiobookReadingLocation
      setLocation:[[TPPBookLocation alloc] initWithLocationString:string renderer:@"NYPLAudiobookToolkit"]
      forIdentifier:self.book.identifier];
     
-    // Save updated location on server
-    [TPPAnnotations postListeningPositionForBook:self.book.identifier selectorValue:string];
+    if ([[NSDate date] timeIntervalSinceDate: self.lastServerUpdate] >= ServerUpdateDelay) {
```
Comment (vladimirfedorov) [inline Palace/Book/UI/TPPBookCellDelegate.m:506]: Does it mean that in worst case scenario the app doesn't save last 15 seconds of progress? I couldn't find if we upload listening position if the user closes the book.

**Review by vladimirfedorov — CHANGES_REQUESTED** [review]

File: Palace/Book/UI/TPPBookCellDelegate.m (hunk 2 above)
Comment (vladimirfedorov) [reply Palace/Book/UI/TPPBookCellDelegate.m:506]: I think saving on exit when the user taps the Back button should be enough

Comment (mauricecarrier7) [reply Palace/Book/UI/TPPBookCellDelegate.m:506]: Resolved here: https://github.com/ThePalaceProject/ios-audiobooktoolkit/pull/35

**Pull request merged**